src/backend/commands/tablecmds.c

```diff
@@ -18362,11 +18362,12 @@ ATPExecPartSplit(Relation *rel,
 			else if (intopid)
 				ErrorOnInvalidDefaultPartition(rel, intopid);
 
-			if (prule->topRule->parisdefault && i == into_exists)
+			if (prule->topRule &&
```

Comment:
I think this doesn't prevent the dereferencing [above](https://github.com/arenadata/gpdb/pull/737/files#diff-d373fb790605aded212c2e3a0fbe8c015d110077ff46c07d0a6d6647a2d7e928R18350).